source/MaterialXRuntime/RtNodeGraph.cpp

```diff
@@ -44,8 +44,9 @@ RtPrim RtNodeGraph::createPrim(const RtToken& typeName, const RtToken& name, RtP
 RtInput RtNodeGraph::createInput(const RtToken& name, const RtToken& type, uint32_t flags)
 {
     PvtPrim* socket = prim()->getChild(SOCKETS);
-    socket->createOutput(name, type, flags | RtAttrFlag::SOCKET);
-    return prim()->createInput(name, type, flags)->hnd();
+    RtInput input = prim()->createInput(name, type, flags)->hnd();
```

Comment:
Unfortunately I need to create the input first as it may have a conflicting child node while the socket won't have one. So this logic needed to be changed a bit.